File: py/decrypt.py
#
#
# Use: decrypt("eqnvz", "XMCKL")
# => "hello"
#
#
def decrypt(st, key):
	if len(st) != len(key):
		return "Text and Key have to be the same length."
	alphabet = list("abcdefghijklmnopqrstuvwxyz")
	nText = []
	kText = []
	for i in range(len(st)):
		nText.append(alphabet.index(st[i].lower()))
		kText.append(alphabet.index(key[i].lower()))
	out = ""
	for i in range(len(nText)):
		op = (nText[i] - kText[i])
		if op < 0:
			x = 26 + op
		else:
			x = op % 26
		out += alphabet[x]
	return out;

File: js/decrypt.js
/*************************

Use: 'eqnvz'.decrypt("XMCKL")
=> "hello"

*************************/
String.prototype.decrypt = function (key) {
	if (this.length != key.length) {
		throw "Text and Key have to be the same length."
		return 0;
	}
	const alphabet = "abcdefghijklmnopqrstuvwxyz".split("")
	let nText = []
	let kText = []
	for (let i of this) {
		nText.push(alphabet.indexOf(i.toLowerCase()))
	}
	for (let i of key) {
		kText.push(alphabet.indexOf(i.toLowerCase()))
	}
	let out = ""
	for (let i in nText) {
		out += alphabet[(nText[i] - kText[i]) < 0 ? 26 + (nText[i] - kText[i]) : (nText[i] - kText[i]) % 26]
	}
	return out;
}


module.exports = (str, key) => str.decrypt(key);
